Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/preprocessor/CqlPreprocessorElmCommonVisitor.kt

```diff
@@ -0,0 +1,902 @@
+                    element is ContextDef ||
+                    element is ExpressionDef
+            ) {
+                val tags = getTags(tree)
+                if (!tags.isEmpty()) {
+                    var a = getAnnotation(element)
+                    if (a == null) {
+                        a = buildAnnotation()
+                        element.annotation.add(a)
+                    }
+                    // If the definition was processed as a forward declaration, the tag processing
+                    // will already
+                    // have occurred
+                    // and just adding tags would duplicate them here. This doesn't account for the
+                    // possibility
+                    // that
+                    // tags would be added for some other reason, but I didn't want the overhead of
+                    // checking for
+                    // existing
+                    // tags, and there is currently nothing that would add tags other than being
+                    // processed from
+                    // comments
+                    if (a.t.isEmpty()) {
+                        a.t.addAll(tags)
+                    }
+                }
+            }
+        } else {
+            if (libraryInfo.definition != null && libraryInfo.headerInterval != null) {
+                val tags = getTags(libraryInfo.header)
+                if (!tags.isEmpty()) {
+                    var a = getAnnotation(libraryBuilder.library)
+                    if (a == null) {
+                        a = buildAnnotation()
+                        libraryBuilder.library.annotation.add(a)
+                    }
+                    a.t.addAll(tags)
+                }
+            }
+        }
+    }
+
+    private fun getTags(header: String?): List<Tag> {
+        var header = header
+        if (header != null) {
+            header = parseComments(header)
+            return parseTags(header)
+        }
+        return emptyList()
+    }
+
+    private fun getTags(tree: ParseTree): List<Tag> {
+        val bi = libraryInfo.resolveDefinition(tree)
+        return if (bi != null) {
+            getTags(bi.header)
+        } else emptyList()
+    }
+
+    private fun parseTags(header: String?): List<Tag> {
+        var header = header
+        header =
+            java.lang.String.join(
+                "\n",
+                Arrays.asList(
+                    *header!!
+                        .trim { it <= ' ' }
+                        .split("\n[ \t]*\\*[ \t\\*]*".toRegex())
+                        .dropLastWhile { it.isEmpty() }
+                        .toTypedArray()
+                )
+            )
+        val tags = ArrayList<Tag>()
+        var startFrom = 0
+        while (startFrom < header.length) {
+            val tagNamePair = lookForTagName(header, startFrom)
+            if (tagNamePair != null) {
+                if (tagNamePair.left.length > 0 && isValidIdentifier(tagNamePair.left)) {
```

Comment:
Instead of length > 0 ! isEmpty()

Reply:
Replaced with `if (tagNamePair.left.isNotEmpty() && isValidIdentifier(tagNamePair.left)) { ... }` https://github.com/cqframework/clinical_quality_language/pull/1450